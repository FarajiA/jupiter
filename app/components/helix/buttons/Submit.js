import React from 'react';
import PropTypes from 'prop-types';

import Button from './Button';

class Submit extends React.Component {
  render() {
    return (
      <Button
        {...this.props}
        submit
      />
    );
  }
}

Submit.propTypes = {
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  classNames: PropTypes.string
};

Submit.defaultProps = {
  disabled: false,
  classNames: 'submit-btn hxPrimary'
};


export default Submit;
