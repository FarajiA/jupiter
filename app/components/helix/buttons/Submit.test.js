import React from 'react';
import renderer from 'react-test-renderer';
import Submit from './Submit';

describe('Submit', () => {
  const props = {
    label: 'Submit!'
  };
  test('it renders correct props', () => {
    const component = renderer.create(<Submit {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Label prop is rendered', () => {
    const wrapper = shallow(<Submit {...props} />);
    expect(wrapper.prop('label')).toEqual('Submit!');
  });
});
