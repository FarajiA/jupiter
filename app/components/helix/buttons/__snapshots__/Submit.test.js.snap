// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit it renders correct props 1`] = `
<button
  className="hxBtn submit-btn hxPrimary"
  disabled={false}
  type="submit"
>
  Submit!
</button>
`;
